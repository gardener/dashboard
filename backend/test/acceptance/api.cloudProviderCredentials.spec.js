// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0

'use strict'

const { mockRequest } = require('@gardener-dashboard/request')
const { Store } = require('@gardener-dashboard/kube-client')
const cache = require('../../lib/cache')

function createStore (items) {
  const store = new Store()
  store.replace(items)
  return store
}

describe('api', function () {
  let agent

  beforeAll(() => {
    agent = createAgent()

    cache.initialize({
      shoots: {
        store: createStore(fixtures.shoots.list()),
      },
    })
  })

  afterAll(() => {
    cache.cache.resetTicketCache()
    return agent.close()
  })

  beforeEach(() => {
    mockRequest.mockReset()
  })

  describe('cloudproviderCredentials', function () {
    const namespace = 'garden-foo'
    const infraName = 'foo-infra3'
    const dnsName = 'foo-dns1'
    // project
    const project = fixtures.projects.getByNamespace(namespace)
    // user
    const id = project.spec.owner.name
    const user = fixtures.auth.createUser({ id })

    it('should return three cloudProvider credentials', async function () {
      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.list())
      mockRequest.mockImplementationOnce(fixtures.secrets.mocks.list())

      const params = {
        secretBindingNamespace: namespace,
      }

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'list', params })
        .expect('content-type', /json/)
        .expect(200)

      expect(mockRequest).toHaveBeenCalledTimes(2)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })

    it('should return no cloudProvider credentials', async function () {
      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.list())
      mockRequest.mockImplementationOnce(fixtures.secrets.mocks.list())

      const namespace = 'garden-baz'
      const params = {
        secretBindingNamespace: namespace,
      }

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'list', params })
        .expect('content-type', /json/)
        .expect(200)

      expect(mockRequest).toHaveBeenCalledTimes(2)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })

    it('should create a cloudProvider infrastructure secret', async function () {
      mockRequest.mockImplementationOnce(fixtures.secrets.mocks.create())
      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.create())

      const params = {
        secretBindingNamespace: namespace,
        secretBindingName: 'new-infra1',
        poviderType: infraName,
        secretData: {
          key: 'myKey',
          secret: 'mySecret',
        },
      }

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'create', params })
        .expect('content-type', /json/)
        .expect(200)

      expect(mockRequest).toHaveBeenCalledTimes(2)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })

    it('should create a cloudProvider dns secret', async function () {
      const params = {
        secretBindingNamespace: namespace,
        secretBindingName: 'new-dns1',
        poviderType: dnsName,
        secretData: {
          key: 'myKey',
          secret: 'mySecret',
        },
      }

      mockRequest.mockImplementationOnce(fixtures.secrets.mocks.create())
      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.create())

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'create', params })
        .expect('content-type', /json/)
        .expect(200)

      expect(mockRequest).toHaveBeenCalledTimes(2)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })

    it('should patch an own cloudProvider credential', async function () {
      const params = {
        secretBindingNamespace: namespace,
        secretBindingName: 'foo-infra1',
        secretData: {
          key: 'myNewKey',
          secret: 'myNewSecret',
        },
      }

      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.get())
      mockRequest.mockImplementationOnce(fixtures.secrets.mocks.patch())

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'patch', params })
        .expect('content-type', /json/)
        .expect(200)

      expect(mockRequest).toHaveBeenCalledTimes(2)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })

    it('should not patch a shared cloudProvider credential', async function () {
      const params = {
        secretBindingNamespace: namespace,
        secretBindingName: 'trial-infra1',
        secretData: {
          key: 'myNewKey',
          secret: 'myNewSecret',
        },
      }

      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.get())

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'patch', params })
        .expect('content-type', /json/)
        .expect(422)

      expect(mockRequest).toHaveBeenCalledTimes(1)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })

    it('should delete an own cloudProvider credential', async function () {
      const params = {
        secretBindingNamespace: namespace,
        secretBindingName: 'foo-infra1',
      }
      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.get())
      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.delete())
      mockRequest.mockImplementationOnce(fixtures.secrets.mocks.delete())

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'remove', params })
        .expect(200)

      expect(mockRequest).toHaveBeenCalledTimes(3)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot()
    })

    it('should not delete a shared cloudProvider credential', async function () {
      const params = {
        secretBindingNamespace: namespace,
        secretBindingName: 'trial-infra1',
      }
      mockRequest.mockImplementationOnce(fixtures.secretbindings.mocks.get())

      const res = await agent
        .post('/api/cloudprovidercredentials')
        .set('cookie', await user.cookie)
        .send({ method: 'remove', params })
        .expect(422)

      expect(mockRequest).toHaveBeenCalledTimes(1)
      expect(mockRequest.mock.calls).toMatchSnapshot()

      expect(res.body).toMatchSnapshot({
        details: expect.any(Object),
      })
    })
  })
})
