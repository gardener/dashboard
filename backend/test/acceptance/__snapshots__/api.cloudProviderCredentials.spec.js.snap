// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api cloudproviderCredentials should create a cloudProvider infrastructure secret 1`] = `
[
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "post",
      ":path": "/api/v1/namespaces/garden-foo/secrets",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
    {
      "apiVersion": "v1",
      "data": {
        "key": "bmV3LWRhdGE=",
      },
      "kind": "Secret",
      "metadata": {
        "name": "new-foo-infra3-secret",
        "namespace": "garden-foo",
      },
      "type": "Opaque",
    },
  ],
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "post",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-foo/secretbindings",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
    {
      "apiVersion": "core.gardener.cloud/v1alpha1",
      "kind": "SecretBinding",
      "metadata": {
        "name": "new-foo-infra3-secretbinding",
        "namespace": "garden-foo",
      },
      "provider": {
        "type": "foo-infra3",
      },
      "secretRef": {
        "name": "new-foo-infra3-secret",
        "namespace": "garden-foo",
      },
    },
  ],
]
`;

exports[`api cloudproviderCredentials should create a cloudProvider infrastructure secret 2`] = `
{
  "quotas": [],
  "secret": {
    "apiVersion": "v1",
    "data": {
      "key": "bmV3LWRhdGE=",
    },
    "kind": "Secret",
    "metadata": {
      "name": "new-foo-infra3-secret",
      "namespace": "garden-foo",
      "resourceVersion": "42",
    },
    "type": "Opaque",
  },
  "secretBinding": {
    "apiVersion": "core.gardener.cloud/v1alpha1",
    "kind": "SecretBinding",
    "metadata": {
      " resourceVersion": "42",
      "name": "new-foo-infra3-secretbinding",
      "namespace": "garden-foo",
    },
    "provider": {
      "type": "foo-infra3",
    },
    "secretRef": {
      "name": "new-foo-infra3-secret",
      "namespace": "garden-foo",
    },
  },
}
`;

exports[`api cloudproviderCredentials should delete an own cloudProvider credential 1`] = `
[
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-foo/secretbindings/foo-infra1",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "delete",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-foo/secretbindings/foo-infra1",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "delete",
      ":path": "/api/v1/namespaces/garden-foo/secrets/secret1",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
]
`;

exports[`api cloudproviderCredentials should delete an own cloudProvider credential 2`] = `{}`;

exports[`api cloudproviderCredentials should not delete a shared cloudProvider credential 1`] = `
[
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-foo/secretbindings/trial-infra1",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
]
`;

exports[`api cloudproviderCredentials should not delete a shared cloudProvider credential 2`] = `
{
  "code": 422,
  "details": Any<Object>,
  "message": "Remove allowed only if secret and secretBinding are in the same namespace",
  "reason": "Unprocessable Entity",
  "status": "Failure",
}
`;

exports[`api cloudproviderCredentials should not patch a shared cloudProvider credential 1`] = `
[
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-foo/secretbindings/trial-infra1",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
]
`;

exports[`api cloudproviderCredentials should not patch a shared cloudProvider credential 2`] = `
{
  "code": 422,
  "details": {
    "name": "UnprocessableEntityError",
    "stack": "UnprocessableEntityError: Patch allowed only if secret and secretBinding are in the same namespace
    at [object Object]
    at [object Object]",
  },
  "message": "Patch allowed only if secret and secretBinding are in the same namespace",
  "reason": "Unprocessable Entity",
  "status": "Failure",
}
`;

exports[`api cloudproviderCredentials should patch an own cloudProvider credential 1`] = `
[
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-foo/secretbindings/foo-infra1",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "put",
      ":path": "/api/v1/namespaces/garden-foo/secrets/secret1",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
    {
      "data": {
        "key": "Zm9vS2V5",
        "secret": "Zm9vU2VjcmV0",
      },
      "metadata": {
        "labels": {
          "cloudprofile.garden.sapcloud.io/name": "infra1-profileName",
        },
        "name": "secret1",
        "namespace": "garden-foo",
      },
    },
  ],
]
`;

exports[`api cloudproviderCredentials should patch an own cloudProvider credential 2`] = `
{
  "quotas": [
    {
      "metadata": {
        "name": "foo-quota1",
        "namespace": "garden-foo",
        "uid": 2,
      },
      "spec": {
        "clusterLifetimeDays": 14,
        "metrics": {
          "cpu": "200",
        },
        "scope": {
          "apiVersion": "v1",
          "kind": "Secret",
        },
      },
    },
    {
      "metadata": {
        "name": "foo-quota2",
        "namespace": "garden-foo",
        "uid": 3,
      },
      "spec": {
        "clusterLifetimeDays": 14,
        "metrics": {
          "cpu": "200",
        },
        "scope": {
          "apiVersion": "v1",
          "kind": "Secret",
        },
      },
    },
  ],
  "secret": {
    "data": {
      "key": "Zm9vS2V5",
      "secret": "Zm9vU2VjcmV0",
    },
    "metadata": {
      "labels": {
        "cloudprofile.garden.sapcloud.io/name": "infra1-profileName",
      },
      "name": "secret1",
      "namespace": "garden-foo",
      "resourceVersion": "43",
    },
  },
  "secretBinding": {
    "kind": "SecretBinding",
    "metadata": {
      "labels": {
        "cloudprofile.garden.sapcloud.io/name": "infra1-profileName",
      },
      "name": "foo-infra1",
      "namespace": "garden-foo",
    },
    "quotas": [
      {
        "name": "foo-quota1",
        "namespace": "garden-foo",
      },
      {
        "name": "foo-quota2",
        "namespace": "garden-foo",
      },
    ],
    "secretRef": {
      "name": "secret1",
      "namespace": "garden-foo",
    },
  },
}
`;

exports[`api cloudproviderCredentials should return no cloudProvider credentials 1`] = `
[
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-baz/secretbindings",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/api/v1/namespaces/garden-baz/secrets?labelSelector=reference.gardener.cloud%2Fsecretbinding%3Dtrue",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
]
`;

exports[`api cloudproviderCredentials should return no cloudProvider credentials 2`] = `
{
  "quotas": [],
  "secretBindings": [],
  "secrets": [],
}
`;

exports[`api cloudproviderCredentials should return three cloudProvider credentials 1`] = `
[
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/apis/core.gardener.cloud/v1beta1/namespaces/garden-foo/secretbindings",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
  [
    {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/api/v1/namespaces/garden-foo/secrets?labelSelector=reference.gardener.cloud%2Fsecretbinding%3Dtrue",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6ImJhckBleGFtcGxlLm9yZyIsImlhdCI6MTU3NzgzNjgwMCwiYXVkIjpbImdhcmRlbmVyIl0sImV4cCI6MzE1NTcxNjgwMCwianRpIjoianRpIn0.7WKy0sNVkJzIqh3QJIF1zk3QjzwFe_zMTv8PmnOCsxg",
    },
  ],
]
`;

exports[`api cloudproviderCredentials should return three cloudProvider credentials 2`] = `
{
  "quotas": [
    {
      "metadata": {
        "name": "foo-quota1",
        "namespace": "garden-foo",
        "uid": 2,
      },
      "spec": {
        "clusterLifetimeDays": 14,
        "scope": {
          "apiVersion": "v1",
          "kind": "Secret",
        },
      },
    },
    {
      "metadata": {
        "name": "foo-quota2",
        "namespace": "garden-foo",
        "uid": 3,
      },
      "spec": {
        "clusterLifetimeDays": 14,
        "scope": {
          "apiVersion": "v1",
          "kind": "Secret",
        },
      },
    },
  ],
  "secretBindings": [
    {
      "kind": "SecretBinding",
      "metadata": {
        "labels": {
          "cloudprofile.garden.sapcloud.io/name": "infra1-profileName",
        },
        "name": "foo-infra1",
        "namespace": "garden-foo",
      },
      "quotas": [
        {
          "name": "foo-quota1",
          "namespace": "garden-foo",
        },
        {
          "name": "foo-quota2",
          "namespace": "garden-foo",
        },
      ],
      "secretRef": {
        "name": "secret1",
        "namespace": "garden-foo",
      },
    },
    {
      "kind": "SecretBinding",
      "metadata": {
        "labels": {
          "cloudprofile.garden.sapcloud.io/name": "infra3-profileName",
        },
        "name": "foo-infra3",
        "namespace": "garden-foo",
      },
      "quotas": [
        {
          "name": "foo-quota1",
          "namespace": "garden-foo",
        },
        {
          "name": "foo-quota2",
          "namespace": "garden-foo",
        },
      ],
      "secretRef": {
        "name": "secret2",
        "namespace": "garden-foo",
      },
    },
    {
      "kind": "SecretBinding",
      "metadata": {
        "labels": {
          "cloudprofile.garden.sapcloud.io/name": "infra1-profileName",
        },
        "name": "trial-infra1",
        "namespace": "garden-foo",
      },
      "quotas": [
        {
          "name": "foo-quota1",
          "namespace": "garden-foo",
        },
        {
          "name": "foo-quota2",
          "namespace": "garden-foo",
        },
      ],
      "secretRef": {
        "name": "trial-secret",
        "namespace": "garden-trial",
      },
    },
    {
      "kind": "SecretBinding",
      "metadata": {
        "labels": {
          "gardener.cloud/dnsProviderName": "foo-dns",
        },
        "name": "foo-dns1",
        "namespace": "garden-foo",
      },
      "quotas": [
        {
          "name": "foo-quota1",
          "namespace": "garden-foo",
        },
        {
          "name": "foo-quota2",
          "namespace": "garden-foo",
        },
      ],
      "secretRef": {
        "name": "secret3",
        "namespace": "garden-foo",
      },
    },
  ],
  "secrets": [
    {
      "data": {
        "key": "Zm9vS2V5",
        "secret": "Zm9vU2VjcmV0",
      },
      "metadata": {
        "labels": {
          "cloudprofile.garden.sapcloud.io/name": "infra1-profileName",
        },
        "name": "secret1",
        "namespace": "garden-foo",
      },
    },
    {
      "data": {
        "key": "Zm9vS2V5",
        "secret": "Zm9vU2VjcmV0",
      },
      "metadata": {
        "labels": {
          "cloudprofile.garden.sapcloud.io/name": "infra3-profileName",
        },
        "name": "secret2",
        "namespace": "garden-foo",
      },
    },
  ],
}
`;
